### DISCUSSION TITLE
LP Phase Processing Fatal Python Error Aborted

### DISCUSSION THREAD

**Discussion by raffyjef** [ARM-DOE/pyart#1205] [Q&A] [answered]
Hi! I am trying to use the LP Phase Processing function and I always encounter this message on the console saying Fatal Python Error Aborted. Do you happen to know what is causing this? Here is what the console flashes:

runfile('/home/jeff/.config/spyder-py3/Phase processing', wdir='/home/jeff/.config/spyder-py3')

##
##
##     JJ Helmus and SM Collis, JORS 2016, doi: 10.5334/jors.119

<frozen importlib._bootstrap>:283: DeprecationWarning: the load_module() method is deprecated and slated for removal in Python 3.12; use exec_module() instead
Unfolding
/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/pyart/correct/phase_proc.py:170: RuntimeWarning: invalid value encountered in sqrt
  _noise = smooth_and_trim(np.sqrt((line - signal) ** 2), window_len=wl)
/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/pyart/correct/phase_proc.py:171: RuntimeWarning: divide by zero encountered in true_divide
  return abs(signal) / _noise
Exec time:  11.790308475494385
Doing  0

Fatal Python error: Aborted

Main thread:
Current thread 0x00007f4cfb51a740 (most recent call first):
  File "/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/pyart/correct/phase_proc.py", line 1132 in phase_proc_lp
  File "/home/jeff/.config/spyder-py3/Phase processing", line 26 in <module>
  File "/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/spyder_kernels/py3compat.py", line 356 in compat_exec
  File "/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/spyder_kernels/customize/spydercustomize.py", line 482 in exec_code
  File "/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/spyder_kernels/customize/spydercustomize.py", line 623 in _exec_file
  File "/home/jeff/anaconda3/envs/wradlib/lib/python3.10/site-packages/spyder_kernels/customize/spydercustomize.py", line 532 in runfile
  File "/tmp/ipykernel_1934/2549763790.py", line 1 in <cell line: 1>

Restarting kernel...
 

Warning: Ignoring XDG_SESSION_TYPE=wayland on Gnome. Use QT_QPA_PLATFORM=wayland to run on Wayland anyway.

**Comment by zssherman** [comment]
Hmm, @raffyjef which solver are you using? Can you share your code for that section? CyLP has been giving us issues of late.

**Comment by raffyjef** [comment]
I assume it uses pyglpk as solver since I did not specify any solver in using the function. Here is my whole code. Thank you

#!/usr/bin/env python3
# -*- coding: utf-8 -*-
"""
Created on Fri Jul 15 13:32:13 2022

@author: jeff
"""

import numpy as np
import matplotlib.pyplot as plt
import pyart


# perform LP phase processing 

filename = 'Copy of TAG201808110830db.uf'
radar = pyart.io.read_uf(filename)

ncp_values = np.ones((radar.nrays, radar.ngates))
ncp = pyart.config.get_metadata('normalized_coherent_power')
ncp['data'] = ncp_values
radar.add_field('normalized_coherent_power', ncp)

# Forcing the first sweep to be processed
radar.sweep_start_ray_index['data'] = np.array([0])

phidp, kdp = pyart.correct.phase_proc_lp(radar, 0.0, debug=True)
radar.add_field('proc_dp_phase_shift', phidp)
radar.add_field('recalculated_diff_phase', kdp)


# Creating plot of the various differential phase fields
display = pyart.graph.RadarDisplay(radar)
fig = plt.figure(figsize=(10, 10))
ax1 = fig.add_subplot(221)
display.plot_ppi('differential_phase', 0, ax=ax1,
                 title='Raw Differential Phase', colorbar_label='',
                 axislabels_flag=False)

ax2 = fig.add_subplot(222)
display.plot_ppi('proc_dp_phase_shift', 0, ax=ax2,
                 title='Processed Differential Phase', colorbar_label='',
                 axislabels_flag=False)

ax3 = fig.add_subplot(223)
display.plot_ppi('specific_differential_phase', 0, ax=ax3,
                 title='Raw Specific Differential Phase', colorbar_label='',
                 axislabels_flag=False)

ax4 = fig.add_subplot(224)
display.plot_ppi('recalculated_diff_phase', 0, ax=ax4,
                 title='Processed Specific Differential Phase',
                 colorbar_label='',
                 axislabels_flag=False)
plt.show()

# plot a fields from a single ray
display = pyart.graph.RadarDisplay(radar)
fig = plt.figure(figsize=[10, 5])
ax = fig.add_subplot(111)
ray_num = 191

# filtered phidp and unfolded phidp
display.plot_ray('proc_dp_phase_shift', ray_num, format_str='b-',
                 axislabels_flag=False, title_flag=False, ax=ax)

display.plot_ray('unfolded_differential_phase', ray_num, format_str='g-',
                 axislabels_flag=False, title_flag=False, ax=ax)

# set labels
ax.set_ylim(0, 250)
ax.set_ylabel('Differential phase shift (degrees)')
ax.set_xlabel('Range (km)')

# plot KDP and reflectivity on second axis
ax2 = ax.twinx()
display.plot_ray('recalculated_diff_phase', ray_num, format_str='r-',
                 axislabels_flag=False, title_flag=False, ax=ax2)
radar.add_field_like('reflectivity', 'scaled_reflectivity',
                     radar.fields['reflectivity']['data']/10.)
display.plot_ray('scaled_reflectivity', ray_num, format_str='b-',
                 axislabels_flag=False, title_flag=False, ax=ax2)

# decorate
ax2.yaxis.grid(color='gray', linestyle='dashed')
ax.legend(display.plots,
          ["Filtered phiDP", "Unfolded phiDP", 'KDP', 'Z/10.0'],
          loc='upper left')
plt.show()

**Comment by raffyjef** [comment, marked as answer]
Hi! I think you right that its a solver's problem. I tried using this:

phidp, kdp = pyart.correct.phase_proc_lp(radar, 0.0, debug=True, LP_solver='pyglpk')

And it worked!

**Comment by zssherman** [comment]
Ah gotcha! Glad you got it to work!

**Comment by raffyjef** [comment]
Thanks for pointing out!
